models/pancakeswap/bnb/pancakeswap_bnb_nft_market_v1_event.sql

```diff
@@ -0,0 +1,89 @@
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_time', 'unique_trade_id']
+    )
+}}
+
+WITH events AS (
+    SELECT 
+        'bnb' AS blockchain,
+        'pancakeswap' AS project,
+        'v1' AS version,
+        evt_block_time AS block_time,
+        tokenId AS token_id,
+        'erc721' AS token_standard,
+        'Single Item Trade' AS trade_type,
+        1 AS number_of_items,
+        'Buy' AS trade_category,
+        buyer AS buyer,
+        seller AS seller,
+        askPrice AS amount_raw,
+        '0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c' currency_contract,
+        'BNB' currency_symbol,
+        collection AS nft_contract_address,
+        contract_address AS project_contract_address,
+        evt_tx_hash AS tx_hash,
+        evt_block_number AS block_number
+    FROM {{source('pancakeswap_v2_bnb','ERC721NFTMarketV1_evt_Trade')}}
+    {% if is_incremental() %}
+    WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+)
+    SELECT 
+        events.blockchain,
+        events.project,
+        events.version,
+        events.block_time,
+        date_trunc('day', events.block_time) AS block_date,
+        events.token_id,
+        bnb_nft_tokens.name collection,
+        events.amount_raw/POWER(10, bnb_bep20_tokens.decimals)*prices.price AS amount_usd,
+        events.token_standard,
+        CASE WHEN agg.name IS NOT NULL THEN 'Bundle Trade' ELSE 'Single Item Trade' END AS trade_type,
+        CAST(events.number_of_items AS DECIMAL(38,0)) AS number_of_items,
+        events.trade_category,
+        'Trade' AS evt_type,
+        events.seller,
+        events.buyer,
+        events.amount_raw/POWER(10, bnb_bep20_tokens.decimals) AS amount_original,
+        CAST(events.amount_raw AS DECIMAL(38,0)) AS amount_raw,
+        COALESCE(events.currency_symbol, bnb_bep20_tokens.symbol) AS currency_symbol,
+        events.currency_contract,
+        events.nft_contract_address,
+        events.project_contract_address,
+        agg.name AS aggregator_name,
+        CASE WHEN agg.name IS NOT NULL THEN agg.contract_address END AS aggregator_address,
+        events.tx_hash,
+        events.block_number,
+        bt.from AS tx_from,
+        bt.to AS tx_to,
+        CAST(0 AS DOUBLE) AS platform_fee_amount_raw,
+        0 AS platform_fee_amount,
+        0 AS platform_fee_amount_usd,
+        CAST(0 AS DOUBLE) AS platform_fee_percentage,
+        CAST(0 AS DOUBLE) AS royalty_fee_amount_raw,
+        0 AS royalty_fee_amount,
+        0 AS royalty_fee_amount_usd,
+        CAST(0 AS DOUBLE) AS royalty_fee_percentage,
+        0 AS royalty_fee_receive_address,
+        0 AS royalty_fee_currency_symbol,
+        events.blockchain || events.project || events.version || events.tx_hash || events.seller  || events.buyer || events.nft_contract_address || events.token_id AS unique_trade_id
```

Comment:
we're trying to minimize the amount of columns used here.
`block_number - tx_hash - evt_index` will do in this case.
```suggestion
        events.block_number || events.tx_hash || events.evt_index AS unique_trade_id
```
You'll have to add the evt_index in the events CTE above.